core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1496,7 +1442,7 @@ final class Stream[+F[_], +O] private[fs2] (private val underlying: Pull[F, O, U
                     emitNonEmpty(acc) ++ startTimeout.flatMap { newTimeout =>
                       go(Chunk.Queue.empty, newTimeout)
                     }
-                  case Left(t) if t != currentTimeout => go(acc, currentTimeout)
```

Comment:
With this guard, we get non-exhaustive pattern match warnings. Similar for all removed guards.